sdk/communication/AzureCommunicationChat/Source/Signaling/CommunicationSignalingClient.swift

```diff
@@ -0,0 +1,195 @@
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import AzureCore
+import Foundation
+import TrouterClientIos
+import UIKit
+
+public class CommunicationSignalingClient {
+    private var selfHostedTrouterClient: SelfHostedTrouterClient
+    private var communicationSkypeTokenProvider: CommunicationSkypeTokenProvider
+    private var trouterUrlRegistrar: TrouterUrlRegistrar
+    private var logger: ClientLogger
+    private var communicationListeners: Set<CommunicationListener> = []
+
+    public init(
+        skypeTokenProvider: CommunicationSkypeTokenProvider,
+        logger: ClientLogger = ClientLoggers.default(tag: "AzureCommunicationSignalingClient")
+    ) {
+        self.communicationSkypeTokenProvider = skypeTokenProvider
+        self.logger = logger
+        let trouterSkypeTokenHeaderProvider = TrouterSkypetokenAuthHeaderProvider(
+            skypetokenProvider: communicationSkypeTokenProvider
+        )
+
+        let communicationCache = CommunicationCache()
+        selfHostedTrouterClient = SelfHostedTrouterClient.create(
+            withClientVersion: getClientVersion(),
+            authHeadersProvider: trouterSkypeTokenHeaderProvider,
+            dataCache: communicationCache,
+            trouterHostname: getTrouterHostname()
+        )
+
+        let regData: TrouterUrlRegistrationData = createRegistrationData()
+        // swiftlint:disable force_cast
+        trouterUrlRegistrar = TrouterUrlRegistrar.create(
+            with: communicationSkypeTokenProvider,
+            registrationData: regData,
+            registrarHostnameAndBasePath: getRegistrarHostnameAndBasePath(),
+            maxRegistrationTtlS: 3600
+        ) as! TrouterUrlRegistrar
+        // swiftlint:enable force_cast
+    }
+
+    public convenience init(
+        token: String
+    ) {
+        let skypeTokenProvider = CommunicationSkypeTokenProvider(skypeToken: token)
+        self.init(skypeTokenProvider: skypeTokenProvider)
+    }
+
+    public func start() {
+        selfHostedTrouterClient.withRegistrar(trouterUrlRegistrar)
+        selfHostedTrouterClient.start()
+    }
+
+    public func stop() {
+        selfHostedTrouterClient.stop()
+        communicationListeners.forEach { listener in
+            selfHostedTrouterClient.unregisterListener(listener)
+        }
+        communicationListeners.removeAll()
+    }
+
+    public func on(event: String, listener: @escaping EventListener) {
```

Comment:
Personally I think this is more swifty if the event was typed as `ChatEventId`
So when you switch case 
```swift
switch event {
case . chatMessageReceived: 
case . typingIndicatorReceived: 
...
}
```

Reply:
oh yah, ChatClient types it as ChatEventId but I forgot to update that here too :)

Reply:
On the name, `ChatEventId` seems odd to me since I think GUID when I think "id" in Azureland. I would suggest `ChatEventType` but I'm not insistent on it.